backend/tests/test_project.py

```diff
@@ -71,3 +71,99 @@ def test_project_routes(self):
             data=json.dumps(update_obj),
         )
         self.assertEqual(response.status_code, 404)
+
+    def test_get_all_types(self):
+        test_list = ["dummy type", "dummy type 2"]
+        p1 = Project(
+            address="dummy address",
+            city="dummy city",
+            province="dummy province",
+            postal_code="dummy postal_code",
+            neighbourhood="dummy neighbourhood",
+            year=2020,
+            name="dummy name",
+            type=test_list[0],
+            contacts=[],
+        )
+        p2 = Project(
+            address="dummy address",
+            city="dummy city",
+            province="dummy province",
+            postal_code="dummy postal_code",
+            neighbourhood="dummy neighbourhood",
+            year=2020,
+            name="dummy name",
+            type=test_list[1],
+            contacts=[],
+        )
+        db.session.add(p1)
+        db.session.add(p2)
+        db.session.commit()
+
+        response = self.client.get("/project/types")
+        self.assertEqual(response.status_code, 200)
+        json_response = json.loads(response.get_data(as_text=True))
+        self.assertEqual(json_response, test_list)
+
+    def test_get_specific_type(self):
+        test_list = ["dummy type", "dummy type 2"]
+        p1 = Project(
+            address="dummy address",
+            city="dummy city",
+            province="dummy province",
+            postal_code="dummy postal_code",
+            neighbourhood="dummy neighbourhood",
+            year=2020,
+            name="dummy name",
+            type=test_list[0],
+            contacts=[],
+        )
+        p2 = Project(
+            address="dummy address",
+            city="dummy city",
+            province="dummy province",
+            postal_code="dummy postal_code",
+            neighbourhood="dummy neighbourhood",
+            year=2020,
+            name="dummy name",
+            type=test_list[1],
+            contacts=[],
+        )
+        p3 = Project(
+            address="dummy address",
+            city="dummy city",
+            province="dummy province",
+            postal_code="dummy postal_code",
+            neighbourhood="dummy neighbourhood",
+            year=2020,
+            name="dummy name",
+            type=test_list[0],
+            contacts=[],
+        )
+        db.session.add(p1)
+        db.session.add(p2)
+        db.session.add(p3)
+        db.session.commit()
+
+        tester = {
+            "address": "dummy address",
+            "city": "dummy city",
+            "province": "dummy province",
+            "postal_code": "dummy postal_code",
+            "neighbourhood": "dummy neighbourhood",
+            "year": 2020,
+            "name": "dummy name",
+            "type": test_list[1],
+            "contacts": [],
+        }
+        # test empty type param
+        response = self.client.get("/project/")
+        self.assertEqual(response.status_code, 404)
```

Comment:
@peterghrong This might not return a 404 error code once #98 is merged as there is an actual route that makes it a valid request. I could be wrong here, I was wondering whether you could check up on this, or show that it will continue to return a 404 status code even if #98 is merged (a screenshot)?

Reply:
Good point Curtis, thanks for pointing this out, I tested it and it seems that it works fine 
![Screen Shot 2021-01-24 at 12 16 15 PM](https://user-images.githubusercontent.com/66083521/105637930-0c2a1580-5e3e-11eb-950c-c788026bb2e0.png)

Reply:
@peterghrong Perfect, tysm!!!! 💯

Reply:
I tried to merge in the master branch and found that `/project/<uuid>` and `/project/<type>` have route conflict. I added a extra param into the url to resolve this so now we have: 
eg. 
- to get all types
`/project/type/types 
`
- to get project by uuid
`/project/id/<uuid>`

Fixed all unittests as well and ran them.
![Screen Shot 2021-01-25 at 11 05 49 AM](https://user-images.githubusercontent.com/66083521/105731314-4ad2d500-5efd-11eb-80d3-570650d2c06e.png)